lib/internal/repl.js

```diff
@@ -153,13 +154,14 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
         if (oldReplJSONHistory) repl.history = JSON.parse(oldReplJSONHistory);
 
         if (!Array.isArray(repl.history)) {
-          throw new Error('Expected array, got ' + typeof repl.history);
+          throw new errors.Error('ERR_INVALID_REPL_HISTORY',
```

Comment:
This should likely be a `TypeError`. Since this is already a semver-major change, may as well fix the type